mlflow/utils/databricks_utils.py

```diff
@@ -846,18 +846,29 @@ def _get_databricks_creds_config(tracking_uri):
     # so that in this function we still have to use
     # configuration providers defined in legacy Databricks CLI python library to
     # read token values.
-    profile, _ = get_db_info_from_uri(tracking_uri)
+    profile, key_prefix = get_db_info_from_uri(tracking_uri)
 
     config = None
 
-    for provider in [
-        _dynamic_token_config_provider,
-        ProfileConfigProvider(profile),
-        SparkTaskContextConfigProvider(),
-        EnvironmentVariableConfigProvider(),
-        DatabricksModelServingConfigProvider(),
-        TrackingURIConfigProvider(tracking_uri),
-    ]:
+    if profile and key_prefix:
+        # legacy way to read credentials by setting `tracking_uri` to 'databricks://scope:prefix'
+        provider_list = [TrackingURIConfigProvider(tracking_uri)]
```

Comment:
```suggestion
        providers = [TrackingURIConfigProvider(tracking_uri)]
```

can we just use the plural form?